core/blockchain.go

```diff
@@ -1657,19 +1724,25 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (chainIndex int, err err
 		stats.processed++
 	}
 
+	ev := ChainInsertEvent{
+		Processed: stats.processed,
+		Queued:    stats.queued,
+		Ignored:   stats.ignored,
+		TxCount:   txcount,
+	}
+	r := &ChainInsertResult{ChainInsertEvent: ev}
```

Comment:
Here's an example of that `Event` wrapping I was talking about for the Insert return types. Kind of nifty, maybe.